detail/core.v1.secret.vue

```diff
@@ -0,0 +1,223 @@
+      return columns;
+    },
+
+    dataRows() {
+      const rows = [];
+      const { data = {} } = this.value;
+
+      Object.keys(data).forEach((key) => {
+        const value = base64Decode(data[key]);
+
+        rows.push({
+          key,
+          value
+        });
+      });
+
+      return rows;
+    },
+
+    dataHeaders() {
+      return [KEY, VALUE];
+    },
+
+    serviceAccountID() {
+      if (this.value.secretType !== SERVICE_ACCT) {
+        return null;
+      } else {
+        const { metadata:{ annotations = {} } } = this.value;
+
+        const uid = annotations[ANNOTATION.SERVICE_ACCT];
+
+        this.findRelatedServices(uid);
+
+        return uid;
+      }
+    },
+
+    serviceSchema() {
+      return this.$store.getters['cluster/schemaFor'](RIO.SERVICE);
+    },
+
+    serviceHeaders() {
+      return [STATE, NAME, SCALE, IMAGE, CREATED];
+    }
+  },
+  watch: {
+    serviceAccountID(neu) {
+      this.findRelatedServices(neu);
+    }
+  },
+  methods: {
+    findServiceAccounts(uid) {
+      return this.$store.dispatch('cluster/findAll', { type: SERVICE_ACCOUNT, opt: { filter: { uid } } }).then((serviceAccounts) => {
+        return serviceAccounts.filter((account) => {
+          return account.metadata.uid === uid;
+        });
+      });
+    },
+
+    async findRelatedServices(uid) {
+      const serviceAccounts = await this.findServiceAccounts(uid);
+
+      if (serviceAccounts) {
+        const serviceIDs = serviceAccounts.reduce((total, account) => {
+          const { ownerReferences = [] } = account.metadata;
+
+          ownerReferences.forEach((reference) => {
+            total.push(reference.uid);
+          });
+
+          return total;
+        }, []);
+
+        const services = await this.$store.dispatch('cluster/findAll', { type: RIO.SERVICE });
+
+        const relatedServices = services.filter((service) => {
+          return serviceIDs.includes(service.metadata.uid);
+        });
+
+        this.relatedServices = relatedServices;
+      }
+    }
+  }
+};
+</script>
+
+<template>
+  <div>
+    <DetailTop :columns="detailTopColumns" />
+    <template v-if="value.secretType===dockerJSON">
+      <SortableTable
+        class="mt-20"
+        key-field="address"
+        :rows="dockerRows"
+        :headers="dockerHeaders"
+        :search="false"
+        :table-actions="false"
+        :row-actions="false"
+      />
+    </template>
+    <template v-else>
+      <div class="mt-20 mb-20">
+        <h4>Values</h4>
+        <SortableTable
+          class="mt-20"
+          :rows="dataRows"
+          :headers="dataHeaders"
+          key-field="value"
+          :search="false"
+          :row-actions="false"
+          :table-actions="false"
+        />
+      </div>
+      <div v-if="relatedServices.length" class="mt-20">
+        <h4>Related Workoads</h4>
+        <ResourceTable :rows="relatedServices" :schema="serviceSchema" :headers="serviceHeaders" :search="false" :show-groups="false" />
```

Comment:
This is weird because it says Workloads (which implies kubernetes pod/deployment/replicaset/etc) but is actually Rio Services (when I might not even have Rio).